go/billing/api.py

```diff
@@ -493,3 +510,32 @@ def getChild(self, name, request):
     def render_GET(self, request):
         request.setResponseCode(200)  # OK
         return ''
+
+    @defer.inlineCallbacks
+    def health_check(self, request):
+        """
+        We want our health check to be comprehensive, so we implement it here.
+        """
+        try:
+            yield self._connection_pool.runQuery("SELECT 1")
+        except (ConnectionDead, psycopg2.InterfaceError):
+            request.setResponseCode(503)
+            request.write("Database connection unavailable\n")
+        else:
+            request.setResponseCode(200)
+            request.write("OK\n")
+        request.finish()
```

Comment:
I'd be happier if the request writing happened on the HealthResource (since it's meant to be handling the request) and this was a function that returned, e.g. ``(200, "OK")`` or ``(503, "Database connection unavailable")``.